## Input data

The core algorithm of SpaNiche requires two input matrices:

-   a cell type abundance matrix

-   a ligand-receptor expression matrix

These two matrices are spatially “smoothed”/“expanded” under different
views. Views represent the spatial ranges at which we interrogate tissue
structure and function, operationalized as different spatial distances
(see the “Cellular composition” and “Cell-cell interactions” subsections
in the “Methods” section of our paper).

## How to obtain the two matrices

Taking spatial transcriptomics data generated by the 10X Visium platform
as an example, running spaceranger produces a spatial folder containing
image and coordinate information, as well as a
filtered\_feature\_bc\_matrix.h5 file containing gene expression
profiles.

The two required input matrices can then be obtained as follows: 1.
Using existing spatial transcriptomics deconvolution tools (e.g.,
cell2location or RCTD), cell type abundance can be inferred for each
spatial spot.

1.  Based on known ligand–receptor interaction databases (from cellchat)
    and expression profiles, the ligand–receptor expression matrix can
    be constructed using built-in functions provided in SpaNiche.

## Detection of spatial colocalization

    library(SpaNiche)
    library(Seurat)
    library(tidyverse)

    spatial.seu=Load10X_Spatial("./example/ST-colon3/")
    spatial.seu=NormalizeData(spatial.seu)
    spatial.seu@meta.data$SB=rownames(spatial.seu@meta.data)
    spatialdf=spatial.seu@images$slice1@coordinates
    spatialdf$col = spatialdf$col / (3 ^ 0.5)

    spot_by_celltype=read.csv("./example/ST-colon3/cell2location_map/res_spotXcelltype.csv",header = T,row.names = 1,check.names = F)

    if (identical(colnames(spatial.seu),rownames(spatialdf))) {
      if(identical(colnames(spatial.seu),rownames(spot_by_celltype))) {
        print("spatial barcode is ok!")
      }
    }

    res.list = spaniche_nmf_celltype_and_lr(
      spatial.seu,
      spatialdf,
      spot_by_celltype,
      smoothing_type = c("view0", "view1", "view2"),
      distance_thre = c(NA, 2/(3^0.5), 4/(3^0.5)),
      LR_dm_type = c("view1", "view2"),
      LR_distance_thre = c(2/(3^0.5), 4/(3^0.5)),
      topic_num = 15,
      defined_weight = c(0.6,0.4),
      lambda_v = 0.5,
      sigma_v = 0.5
    )

## Exploration of Results and Visualization

**Note:** SpaNiche currently relies on Seurat v4 internally. Support for
Seurat v5 will be added in future releases.
